### DISCUSSION TITLE
Installation Error: bin/console kimai:install -n

### DISCUSSION THREAD

**Discussion by JohniAKim** [kimai/kimai#3186] [On-Premise: Installation & Updates]
**Describe the bug**
Attempting to install on TurnkeyLinux LAMP appliance.  Receive an error running the following:
`bin/console kimai:install -n`

Producing the following:
```
root@KIMAI-01 .../www/kimai2# bin/console kimai:install -n
PHP Warning:  require(/var/www/kimai2/vendor/autoload.php): failed to open stream: No such file or directory in /var/www/kimai2/bin/console on line 12
PHP Stack trace:
PHP   1. {main}() /var/www/kimai2/bin/console:0
PHP Fatal error:  require(): Failed opening required '/var/www/kimai2/vendor/autoload.php' (include_path='.:/usr/share/php') in /var/www/kimai2/bin/console on line 12
PHP Stack trace:
PHP   1. {main}() /var/www/kimai2/bin/console:0
```

I was unable to locate a vendor folder nor an autoload.php file

**To Reproduce**
Install as per https://www.kimai.org/documentation/installation.html

**Logfile**
```
Add the last lines from your logfile at "var/log/prod.log" or "Doctor > Logs", around the time when the problem happened.
```
NO LOG FILE FOUND

**Additional context**
 - Kimai version: 1.18.1 
 - PHP version: 7.3
 - Device: VirtualBox running TurnkeyLinux LAMP 16.1 appliance (Debian 10)
 - Browser N/A

**Comment by kevinpapst** [comment]
You did not follow the instructions, otherwise it would work 😁 

This one is missing:
```
composer install --no-dev --optimize-autoloader -n
```

**Reply by JohniAKim** [reply]
I ran that step but the VirtualBox terminal ran off screen and I did not see there was a problem with a composer lock file not containing a compatible set of packages.  I ran `composer update` but still have several problems that composer is complaining about.

Geez, all you want to do is test drive something and now I am running around replacing car parts.  Reminds me of DLL hell.

I will work on fixing composer and see if that resolves my issue.  I will follow up afterwards.

**Reply by kevinpapst** [reply]
That sounds as if your PHP installation is incompatible. Either missing PHP modules or an incompatible PHP version (8.1 is not yet supported).

**Comment by JohniAKim** [comment]
Running PHP 7.3.  I had to install some packages (php-gd, php-mbstring, and php-intl).

Was able to successfully run:
`composer install --no-dev --optimize-autoloader -n`

And then successfully ran:
bin/console kimai:install -n

Now I have go and figure out setting up this thing called Apache Server so I can actually access KIMAI via browser.  Any tips for somebody that has never touched the stuff.

BTW, thank you for the quick responses.  It is greatly appreciated.

**Reply by kevinpapst** [reply]
Oh well, if you have never setup a Webserver, this might get dirty... there is a documentation related to the required Webserver configuration at https://www.kimai.org/documentation/webserver-configuration.html

But this kind of assumes that it is already running and pre-configured to support mod-rewrite and php.

I can help with Kimai, but not with setting up your environment. That question is better suited at places like https://serverfault.com

**Reply by JohniAKim** [reply]
Ok, got it setup and running.  Because I am using a Turnkeylinux appliancea that uses Postfix configured with SendInBlue, I had to figure out a couple of things for the email.  Syntax I needed was 'MAILER_URL=sendmail://default'.

As far as setting up the Apache server which came with the appliance, I found a good Youtube video that really walks through it all pretty well.

In case anybody else needs the help the video is at [Apache Virtual Hosts](https://www.youtube.com/watch?v=x5fWSWdM4F8)
